boa/test/plugin.py

```diff
@@ -27,13 +27,129 @@ def f(*args, **kwargs):
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "ignore_isolation")
+    config.addinivalue_line("markers", "profile_calls")
+
+    pytest.call_profile = {}
+
+
+def _items_to_profile(item):
+
+    profile_test = item.get_closest_marker("profile_calls") is not None
+    if not profile_test:
+        return False
+    return True
+
+
+def _enable_call_profiling(item):
+
+    if not _items_to_profile(item):
+        return
+
+    profile_calls = item.get_closest_marker("profile_calls").args
+    if len(profile_calls) > 0:
+
+        for call in profile_calls:
+            fixture_name = call.split(".")[0]
+            for name, fixture in item.funcargs.items():
+
+                if name == fixture_name:
+
+                    fixture.profile_calls = True
```

Comment:
this has a couple problems. first of all, `profile_calls` is not unset at the end of the item. second of all, the approach is brittle. it relies on parsing the pytest arguments at runtime. what if there is a collision between fixture names and called function names?

it's going to be much cleaner if `enable_call_profiling` sets a flag globally (or on `boa.env`) which can be checked by all contracts at execution, and it's easier to reason about the state of the flag (set/unset) and has less dependency to set up and teardown, since the flag is either set or unset.